### ISSUE TITLE
OOM Errors on Single-Node A800 (80GB) Even with DeepSpeed Zero-3 Offload – Seeking Optimal Hyperparameters

### ISSUE THREAD

**Issue by fengzehui0422** [TianheWu/VisualQuality-R1#7]
I’m encountering persistent Out-of-Memory (OOM) errors when training a multimodal model (Qwen2.5-VL-7B-Instruct) on a single A800 GPU (80GB VRAM). Below is the detailed context and troubleshooting steps I’ve tried, followed by my core question on hyperparameter tuning.
1. Training Setup
GPU: Single NVIDIA A800 (80GB VRAM)
Model: Qwen2.5-VL-7B-Instruct (multimodal, 7B parameters)
Framework: PyTorch + DeepSpeed (Zero-3 with offload configured)
Key Training Command Snippet:
bash
torchrun --nproc_per_node="1" \
    --master_addr="127.0.0.1" --master_port="12345" \
    grpo_jsonl.py \
    --deepspeed local_scripts/zero3.json \  # Zero-3 with optimizer offload to CPU
    --model_name_or_path Qwen2.5-VL-7B-Instruct \
    --num_generations 6 \                   # Initial value (OOM)
    --per_device_train_batch_size 48 \      # Initial value (OOM)
    --gradient_accumulation_steps 1 \       # Initial value
    --bf16 \
    --gradient_checkpointing true \
    --attn_implementation flash_attention_2
2. Core Question
For a single A800 (80GB) running Qwen2.5-VL-7B-Instruct with DeepSpeed Zero-3 (optimizer offload) + BF16 + gradient checkpointing + FlashAttention-2:What is the maximum stable combination of the following hyperparameters that avoids OOM?
--num_generations (number of generations per prompt)
--per_device_train_batch_size (batch size per GPU)
--gradient_accumulation_steps (steps to accumulate gradients)
Are there any known benchmarks or rules of thumb for 7B multimodal models on A800 (80GB) to balance these parameters?
3. Additional Context
The model includes visual encoders (multimodal), which add extra VRAM overhead compared to text-only 7B models.
I’ve verified no other processes are using GPU memory (nvidia-smi shows 0% usage pre-training).
DeepSpeed Zero-3 config: stage: 3, offload_optimizer: {device: "cpu"}, offload_parameters: {device: "cpu"} (tried both with/without parameter offload – parameter offload caused slower training but still OOM).

**Comment by TianheWu** [comment]
Thanks for raising this! You may try reducing the input image resolution, since high resolutions can significantly increase memory consumption. Another option is to lower --per_device_train_batch_size, but please make sure that its value remains a multiple of --num_generations.

**Comment by fengzehui0422** [comment]
@TianheWu Thankyou!  And can you realize the evaluation code and metrics code?  Or maybe It has been realized, but I try my best to find it. I didn't find them !

**Comment by xiangliustc** [comment]
> [@TianheWu](https://github.com/TianheWu) Thankyou! And can you realize the evaluation code and metrics code? Or maybe It has been realized, but I try my best to find it. I didn't find them !

hello, do you achieve running the code? I want to run this code with 4 a800 80g can i avoid oom?

**Comment by TianheWu** [comment]
> [@TianheWu](https://github.com/TianheWu) Thankyou! And can you realize the evaluation code and metrics code? Or maybe It has been realized, but I try my best to find it. I didn't find them !

I have released the label of test set. The validation code is easy to implement with the help of GenAI. Thanks.

**Comment by TianheWu** [comment]
> > [@TianheWu](https://github.com/TianheWu) Thankyou! And can you realize the evaluation code and metrics code? Or maybe It has been realized, but I try my best to find it. I didn't find them !
> 
> hello, do you achieve running the code? I want to run this code with 4 a800 80g can i avoid oom?

Try to reduce the resolution (512x384 is worked in a800 80g) or batchsize

**Comment by xiangliustc** [comment]
> > > [@TianheWu](https://github.com/TianheWu) Thankyou! And can you realize the evaluation code and metrics code? Or maybe It has been realized, but I try my best to find it. I didn't find them !
> > 
> > 
> > hello, do you achieve running the code? I want to run this code with 4 a800 80g can i avoid oom?
> 
> Try to reduce the resolution (512x384 is worked in a800 80g) or batchsize

do u mean it can work on one single a800 80g or 8*a800 80g because i have checked that the resolution is 512x384 and batchsize is 6 (equal to num_generations) but i still meet oom question, or do you have special  process to deal with resolution that i don't find ?thank u for your help
